specutils/spectra/spectrum1d.py

```diff
@@ -134,6 +134,13 @@ def energy(self):
     def bin_edges(self):
         return self.wcs.bin_edges()
 
+    @property
+    def shape(self):
+        return (self.spectral_axis.size, self.flux.size)
```

Comment:
I would have said this should be ``self.flux.shape``.  Isn't that more consistent with ``__len__`` being `self.flux.size`?

That is the "thing" that has a shape is the flux.  The spectral axis' shape is always the same as the leading dimension of `flux` or else something is broken.